src/api/pipelineRuns.js

```diff
@@ -170,30 +173,46 @@ export function createPipelineRun({
   return post(uri, payload).then(({ body }) => body);
 }
 
-export function rerunPipelineRun(pipelineRun) {
-  const { annotations, labels, name, namespace } = pipelineRun.metadata;
+export function generateNewPipelineRunPayload(pipelineRun, rerun) {
+  const { annotations, labels, name, namespace, generateName } =
+    pipelineRun.metadata;
 
   const payload = deepClone(pipelineRun);
   payload.apiVersion =
     payload.apiVersion || `tekton.dev/${getTektonPipelinesAPIVersion()}`;
   payload.kind = payload.kind || 'PipelineRun';
+
+  function getGenerateName() {
+    if (rerun) {
+      return getGenerateNamePrefixForRerun(name);
+    }
+
+    return generateName || getGenerateNamePrefixForNewRun(name);
+  }
+
   payload.metadata = {
-    annotations: annotations || {},
-    generateName: getGenerateNamePrefixForRerun(name),
-    labels: {
-      ...labels,
-      'dashboard.tekton.dev/rerunOf': name
-    },
+    annotations,
+    generateName: getGenerateName(),
+    labels,
     namespace
   };
+  if (rerun) {
+    payload.metadata.labels = {
+      ...labels,
+      'dashboard.tekton.dev/rerunOf': name
+    };
```

Comment:
Let's keep the `...labels` above to handle potentially missing labels, set defaults, etc.

Then here for the rerun, just add the one specific to reruns:

`payload.metadata.labels['dashboard.tekton.dev/rerunOf'] = name;`

This approach keeps clear separation between the two and makes it easier to refactor in future.